base/sparse/sparsematrix.jl

```diff
@@ -1884,48 +1884,61 @@ function setindex!{Tv,Ti,T<:Real}(A::SparseMatrixCSC{Tv,Ti}, x, I::AbstractVecto
 end
 
 
-
 # Sparse concatenation
 
 function vcat(X::SparseMatrixCSC...)
+    Tv = promote_type(map(x->eltype(x.nzval), X)...)
+    Ti = promote_type(map(x->eltype(x.rowval), X)...)
+
+    vcat(map(x->convert(SparseMatrixCSC{Tv,Ti}, x), X)...)
```

Comment:
Converting is a pretty uncommon case, which is why I thought that it was worth the simplicity to just convert. Anyways, I tried implementing `stuffcol!` (code below in comments), but that is still slower than my current implementation, and takes more memory.